### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by neilcook**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -13,7 +13,12 @@ PKG_CHECK_MODULES([GETDNS], [getdns], [AC_DEFINE([HAVE_GETDNS], [1], [Define to
 PKG_CHECK_MODULES([libsodium], [libsodium], [AC_DEFINE([HAVE_LIBSODIUM], [1], [Define to 1 if you have libsodium])])
 PKG_CHECK_MODULES([LIBSYSTEMD], [libsystemd], [AC_DEFINE([HAVE_LIBSYSTEMD], [1], [Define to 1 if you have libsystemd-dev])], [true])
 PKG_CHECK_MODULES([LIBHIREDIS], [libhiredis], [AC_DEFINE([HAVE_LIBHIREDIS], [1], [Define to 1 if you have libhiredis-dev])], [true])
-PKG_CHECK_MODULES([LIBHIREDIS], [hiredis], [AC_DEFINE([HAVE_LIBHIREDIS], [1], [Define to 1 if you have hiredis])], [true])
+AS_IF([test "x$LIBHIREDIS_LIBS" == "x"], [
+  PKG_CHECK_MODULES([LIBHIREDIS], [hiredis], [AC_DEFINE([HAVE_LIBHIREDIS], [1], [Define to 1 if you have hiredis])])
+])
+AS_IF([test "x$LIBHIREDIS_LIBS" == "x"], [
+  AC_MSG_ERROR([libhiredis not found, libhiredis support is not optional])
```
Comment (cmouse) [inline configure.ac]: this could be "libhiredis is required", as it could be elsewhere.

File: wforce_exception.hh [hunk 2]
```diff
@@ -28,3 +28,6 @@ class WforceException: public std::runtime_error
 public:
   WforceException(const std::string &a) : std::runtime_error(a) {}
 };
+
+// This means we can use PDNS classes that emit PDNSException
+typedef WforceException PDNSException;
```
Comment (cmouse) [inline wforce_exception.hh]: Why not just rename, unless you are actually using powerdns sources directly?

File: configure.ac (hunk 1 above)
Comment (neilcook) [inline configure.ac]: Yes good point, should make it standard

File: wforce_exception.hh (hunk 2 above)
Comment (neilcook) [inline wforce_exception.hh]: Well there are a couple of places where I use pdns sources directly, so this is "insurance". It makes merging changes a lot easier as well.

**Pull request merged**